src/rgw/rgw_cache.h

```diff
@@ -181,12 +181,9 @@ class RGWCache  : public T
   }
 
   string normal_name(rgw_pool& pool, const std::string& oid) {
-    string& pool_name = pool.name;
-    char buf[pool_name.size() + 1 + oid.size() + 1];
-    const char *pool_str = pool_name.c_str();
-    const char *oid_str = oid.c_str();
-    sprintf(buf, "%s+%s", pool_str, oid_str);
-    return string(buf);
+    std::string buf;
+    buf.reserve(pool.name.size() + oid.size() + 2);
+    return buf.append(pool.name).append("+").append(oid);
```

Comment:
oops sorry, this last line will interfere with [return value optimization](https://en.wikipedia.org/wiki/Return_value_optimization), and induce an extra alloc/copy back to the caller. splitting up the `append()`s and `return` will help:
```c++
    buf.append(pool.name).append("+").append(oid);
    return buf;
```